src/main.cpp

```diff
@@ -6792,6 +6792,7 @@ void ProcessTxBaseMsg(const CTransactionBase& txBase, CNode* pfrom)
 
     pfrom->setAskFor.erase(inv.hash);
     mapAlreadyAskedFor.erase(inv);
+    mapAlreadyReceived.insert(std::make_pair(inv, GetTimeMicros()));
```

Comment:
This is a good question. In principle, once you have and `inv` in the map you should not execute `ProcessTxBaseMsg` anymore. Still, this could happen if a peer's `tx` response is delayed by a lot, so that you first send out multiple requests for the same `inv`, and eventually receive responses for all the requests.

Still, the `mapAlreadyReceived`'s size is thought to accommodate a constant flow of 1000tx/s for 16min, i.e. the maximum expected delay for the execution of an inv request (considering the default parameters of 8 peers).
So, I'd rather not update entries after the initial insertion, if additional requests are actually performed. The update would keep the entry around longer than needed, and use a slot which might be better used by a more recent `inv` request.